source/models/row.py

```diff
@@ -1,15 +1,19 @@
 import os, json, uuid
-import time
 from outputs.exceptions import *
 from models.index import Index
 
 class Row:
-    def __init__(self, table, value = None):
+    def __init__(self, table, value = {}):
         self.table = table
         self.value = value
-        if self.table.table_metadata.primary_key not in value : 
-            value[self.table.table_metadata.primary_key] = str(uuid.uuid4().node)
-        self.primary_key = value[self.table.table_metadata.primary_key]
+        self.primary_key = self.__get_primary_key()
+
+    def __get_primary_key(self):
+        if self.table.table_metadata.primary_key not in self.value : 
+            self.value[self.table.table_metadata.primary_key] = str(uuid.uuid4().node)
```

Comment:
You shouldn’t update value here, you should update this in constructor

Reply:
He should deal with row object. Value is hidden from him

Reply:
i removed this function and made the check in the constructor.
done